ChallengeUI/src/main/java/com/drivequant/drivekit/challenge/ui/viewmodel/ChallengeListViewModel.kt

```diff
@@ -0,0 +1,92 @@
+package com.drivequant.drivekit.challenge.ui.viewmodel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.drivequant.drivekit.challenge.ChallengesQueryListener
+import com.drivequant.drivekit.challenge.ChallengesSyncStatus
+import com.drivequant.drivekit.challenge.DriveKitChallenge
+import com.drivequant.drivekit.core.DriveKit
+import com.drivequant.drivekit.databaseutils.entity.Challenge
+import com.drivequant.drivekit.databaseutils.entity.ChallengeStatus
+
+class ChallengeListViewModel : ViewModel() {
+
+    private var challengeListData = mutableListOf<ChallengeData>()
+    var mutableLiveDataChallengesData: MutableLiveData<List<ChallengeData>> = MutableLiveData()
+    var selectedChallengeStatusData: ChallengeStatusData
+    var challengesStatusData = mutableListOf<ChallengeStatusData>()
+    var filteredChallenge = mutableListOf<ChallengeData>()
+    var syncChallengesError: MutableLiveData<Any> = MutableLiveData()
+        private set
+
+    init {
+        challengesStatusData.add(
+            ChallengeStatusData(
+                "dk_challenge_active",
+                listOf(ChallengeStatus.PENDING, ChallengeStatus.SCHEDULED)
+            )
+        )
+        challengesStatusData.add(
+            ChallengeStatusData(
+                "dk_challenge_finished",
+                listOf(ChallengeStatus.FINISHED, ChallengeStatus.ARCHIVED)
+            )
+        )
+        selectedChallengeStatusData = challengesStatusData.first()
+    }
+
+    fun filterChallenges(statusList: List<ChallengeStatus>) {
+        filteredChallenge.clear()
+        for (challengeData in challengeListData) {
+            for (status in statusList) {
```

Comment:
Voir si on peut remplacer ce for imbriqué par `if (statusList.contains(challengeData.status)){ ... }`